Bluetooth: Mesh: Rename prov_dev->provisionee

Andrewpini opened the pull request:
Renames "prov device" references and options to
"provisionee" to align implementation with Mesh
Protocol specification v1.1, section 5.4.

Fixes #64315

carlescufi commented on doc/releases/release-notes-3.6.rst:
```diff
@@ -52,6 +52,10 @@ Bluetooth
 
 * Mesh
 
+  * Deprecated :kconfig:option:`CONFIG_BT_MESH_PROV_DEVICE`. This option is
```
This belongs in the migration guide instead

PavelVPV commented on doc/releases/migration-guide-3.6.rst:
```diff
@@ -32,6 +32,10 @@ Bootloader
 Bluetooth
 =========
 
+  * Mesh: Deprecated :kconfig:option:`CONFIG_BT_MESH_PROV_DEVICE`. This option is
```
Each change should be on a new line, example:
https://github.com/zephyrproject-rtos/zephyr/blob/bfff0f9f4cdf8f32454c87b0c15a16be4ca1fc4f/doc/releases/release-notes-3.5.rst?plain=1#L201-L204

PavelVPV commented on subsys/bluetooth/mesh/provisionee.c:
```diff
@@ -698,8 +698,8 @@ int bt_mesh_prov_enable(bt_mesh_prov_bearer_t bearers)
 
-#if defined(CONFIG_BT_MESH_PROV_DEVICE_LOG_LEVEL)
-	if (CONFIG_BT_MESH_PROV_DEVICE_LOG_LEVEL > 2) {
+#if defined(CONFIG_BT_MESH_PROVISIONEE_LOG_LEVEL)
```
We need a follow up task to use the new option in nRF Connect SDK: https://github.com/nrfconnect/sdk-nrf/blob/e29e9d986680d3ddf691f7539b017d2210ae9600/samples/bluetooth/mesh/dfu/distributor/prj.conf#L83

Andrewpini replied:
I've created a task for it 👍

Andrewpini committed:
Bluetooth: Mesh: Rename prov_dev->provisionee

Renames "prov device" references and options to
"provisionee" to align implementation with Mesh
Protocol specification v1.1, section 5.4.

Signed-off-by: Anders Storrø <anders.storro@nordicsemi.no>
